src/plugins/workspace/public/utils.ts

```diff
@@ -267,18 +291,37 @@ export const getDirectQueryConnections = async (dataSourceId: string, http: Http
 };
 
 export const convertDataSourcesToOpenSearchConnections = (
-  dataSources: DataSource[]
+  dataSources: DataConnection[] | DataSource[]
 ): DataSourceConnection[] =>
-  dataSources.map((ds) => {
+  dataSources
+    .filter((ds) => ds.type === DATA_SOURCE_SAVED_OBJECT_TYPE)
+    .map((ds: DataSource) => {
+      return {
+        id: ds.id,
+        type: ds.dataSourceEngineType,
+        connectionType: DataSourceConnectionType.OpenSearchConnection,
+        name: ds.title,
+        description: ds.description,
+        relatedConnections: [],
+      };
+    });
+
+export const convertDataSourcesToDataConnections = (
```

Comment:
Nit: maybe can consider to combine `convertDataSourcesToOpenSearchConnections` and `convertDataSourcesToDataConnections`

Reply:
Updated.